src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,315 @@
+      case 'tertiary':
+        return '0px';
+      default:
+        return '0px';
+    }
+  },
+  underlayColor({ variant, variantColor, theme }) {
+    switch (variant) {
+      case 'primary':
+        return getColor(theme, `${variantColor || 'primary'}.600`);
+      case 'secondary':
+        return getColor(theme, `${variantColor || 'primary'}.950`);
+      case 'tertiary':
+        return getColor(theme, `${variantColor || 'primary'}.950`);
+      default:
+        return getColor(theme, `${variantColor || 'primary'}.600`);
+    }
+  },
+  padding({ size, theme, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return [theme.spacings.xxsmall];
+        }
+        return [theme.spacings.xxsmall, theme.spacings.small];
+      case 'small':
+        if (!children) {
+          return [makePxValue(0.75)];
+        }
+        return [makePxValue(0.625), theme.spacings.large];
+      case 'medium':
+        if (!children) {
+          return [theme.spacings.small];
+        }
+        return [theme.spacings.small, theme.spacings.xxlarge];
+      case 'large':
+        if (!children) {
+          return [makePxValue(1.25)];
+        }
+        return [makePxValue(1.25), theme.spacings.xxlarge];
+      default:
+        if (!children) {
+          return [theme.spacings.small];
+        }
+        return [theme.spacings.small, theme.spacings.xxlarge];
+    }
+  },
+  iconSize({ size, children }) {
+    switch (size) {
+      case 'xsmall':
+        if (!children) {
+          return 'small';
+        }
+        return 'xsmall';
+      case 'small':
+        if (!children) {
+          return 'medium';
+        }
+        return 'small';
+      case 'medium':
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+      case 'large':
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+      default:
+        if (!children) {
+          return 'large';
+        }
+        return 'medium';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        return iconAlign === 'left' ? [0, 0.5, 0, 0] : [0, 0, 0, 0.5];
```

Comment:
let's follow one convention? some places you have use `if` conditions some places it's ternary operator. Let's stick to `if` in styles cases